### DISCUSSION TITLE
I/O operation on closed file - Firefox

### DISCUSSION THREAD

**Discussion by slvamegestic** [seleniumbase/SeleniumBase#2473] [Q&A] [answered]
```
from seleniumbase import SB

def run_test():
    sb_config = {'browser': 'firefox'}
    with SB(**sb_config) as sb:
        sb.maximize_window()
        sb.driver.get('https://www.example.com')

if __name__ == "__main__":
    run_test()
```
Hi all, above is an example of the code that I have shared, Above is working perfectly for Chrome and Safari browsers. but for firefox i am getting below error
```
selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1
...
ValueError: I/O operation on closed file

```
I am wondering why this only happens for Firefox.
My Firefox version is: Version 115.7.0esr (64-bit)

any help on how can I overcome this? also suggest to me how to use different versions of browsers using CLI

**Comment by mdmintz** [comment, marked as answer]
Could be related to https://github.com/SeleniumHQ/selenium/issues/12487
Also could be that you need to use a regular Firefox, instead of a beta one.
If neither of those helped, try using `headless` mode, or a virtual display.
